handbook/engineering/distribution/observability/monitoring_architecture.md

```diff
@@ -0,0 +1,118 @@
+
+*Rationale for Alertmanager*: An approach for notifiers using Grafana was considered, but had some issues outlined in [#11832](https://github.com/sourcegraph/sourcegraph/pull/11832), so Alertmanager was selected as our notification provider.
+
+*Rationale for silencing in site configuration*: Similar to the [Grafana admin reverse-proxy](#admin-reverse-proxy), silencing using the Alertmanager UI would require port-forwarding, something we [want to avoid](monitoring_pillars.md#long-term-vision).
+
+#### prom-wrapper
+
+The [prom-wrapper](https://github.com/sourcegraph/sourcegraph/tree/master/docker-images/prometheus/cmd/prom-wrapper) is the entrypoint program for `sourcegraph/prometheus` and it:
+
+* Handles starting up Prometheus and Alertmanager
+* Applies updates to site configuration by [generating a diff and applying changes](https://sourcegraph.com/search?q=repo:%5Egithub.com/sourcegraph/sourcegraph%24+file:docker-images/prometheus+type:symbol+Change+OR+Diff&patternType=literal)
+  * Most notably, this includes [configuring notifiers and silences](#alert-notifications) for Sourcegraph alerts
+* Exposes [endpoints for configuration issues, alerts summary statuses, and reverse-proxies Prometheus and Alertmanager](https://sourcegraph.com/search?q=repo:%5Egithub.com/sourcegraph/sourcegraph%24+file:docker-images/prometheus+PathPrefix%28:%5Bpath%5D%29.Handler%28:%5Bhandler%5D%29&patternType=structural)
+
+*Rationale for an all-in-one Prometheus image with prom-wrapper*: This allows us to avoid adding a new separate service (which must be handled in all our deployment types), thus simplifying the deployment story for our monitoring stack, while also improving the alert debugging workflow (for example, simply port-forward `svc/prometheus` to get access to the entire alerting stack), with minimal disadvantages (for example, high-availability Prometheus and Alertmanager can still be configured via `PROMETHEUS_ADDITIONAL_FLAGS` and `ALERTMANAGER_ADDITIONAL_FLAGS`, and Alertmanager can be disabled via `DISABLE_ALERTMANAGER`).
+
+## Custom additions
+
+TODO: how we handle out-of-band metrics, alerts (things we don't ship to customers)
+
+## Sourcegraph Cloud
+
+This section describes how our monitoring stack is used in Sourcegraph Cloud and what customizations we have made.
+
+### Alerts
+
+Notifiers for alerts are configured via the [`deploy-sourcegraph-dot-com` frontend ConfigMap for `site.json`](https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/base/frontend/sourcegraph-frontend.ConfigMap.yaml#L5188-L5274).
+
+Alerts go to the [`#alerts`](https://sourcegraph.slack.com/archives/CSCFMFXS5)channel, with critical alerts going to [OpsGenie](https://about.sourcegraph.com/handbook/engineering/on_call).
+
+### Blackbox exporter
+
+TODO
```

Comment:
@davejrt is there a quick writeup/summary/link somewhere about blackbox I can add here? If you think it's a lot I could remove this section for now and have it added in a separate PR as well :)

Reply:
@bobheadxi Let me know if that description is sufficient for this doc

Reply:
looks great, thank you for adding this! :)